test/functional/tool_wallet.py

```diff
@@ -167,6 +167,42 @@ def test_tool_wallet_info(self):
         assert_equal(shasum_before, shasum_after)
         self.log.debug('Wallet file shasum unchanged\n')
 
+    def test_tool_wallet_dump_readonly(self):
```

Comment:
In 6189a20491f71e3a6f499d2a510b38bb914e00d9 "test: Add tests for readonly database access"

There is a lot of code duplication with `test_tool_wallet_info`, it'd be nice if this could be de-deuplicated.